### PR TITLE
Use a unique origin name to avoid poorly configured git

### LINKED ISSUE #1403: [corner case?] pre-commit fails to add origin when there's settings under origin in the user-global config
Basically, if you do
```console
git config --global --add remote.origin.fetch '+refs/pull/*/head:refs/remotes/origin/pr/*'
```
it effectively adds 
```ini
[remote "origin"]
	fetch = +refs/pull/*/head:refs/remotes/origin/pr/*
```
to `~/.gitconfig`.

And then the failure looks like this:
```console
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.git.
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/git', 'remote', 'add', 'origin', 'https://github.com/pre-commit/pre-commit-hooks.git')
return code: 128
expected return code: 0
stdout: (none)
stderr:
    fatal: remote origin already exists.
    
Check the log at ~/.cache/pre-commit/pre-commit.log
```

The solution would be to check if the remote exists and maybe instead do
```console
git config --local remote.origin.url 'https://github.com/pre-commit/pre-commit-hooks.git'
git config --local remote.origin.fetch '+refs/heads/*:refs/remotes/origin/*'
git fetch --all
```

### PR DESCRIPTION
**Pull request by asottile**
Resolves #1403

### CODE REVIEW

File: pre_commit/git.py [hunk 1]
```diff
@@ -10,9 +10,10 @@
 from pre_commit.util import cmd_output_b
 from pre_commit.util import EnvironT
 
-
 logger = logging.getLogger(__name__)
 
+ORIGIN = '_pre-commit-origin'
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply pre_commit/git.py:15]: I think it'd be more accurate to call it `REMOTE` I guess.

[the remark this answers is not in the document]
Comment (asottile) [reply pre_commit/git.py:15]: it's a replacement for `'origin'` so I think it's correct as is

**Comment by webknjaz** [comment]
@asottile how about using `GIT_CONFIG_NOSYSTEM` and `GIT_CONFIG_NOGLOBAL` env vars instead? This'd ultimately improve the consistency of any Git commands you run.

https://superuser.com/a/58287/182499

**Comment by asottile** [comment]
> 
> 
> @asottile how about using `GIT_CONFIG_NOSYSTEM` and `GIT_CONFIG_NOGLOBAL` env vars instead? This'd ultimately improve the consistency of any Git commands you run.
> 
> https://superuser.com/a/58287/182499

see: https://github.com/pre-commit/pre-commit/issues/1403#issuecomment-616692028

**Comment by asottile** [comment]
this broke `go` in a backward-incompatible way so I'm adjusting the approach

File: pre_commit/git.py [hunk 2]
```diff
@@ -158,8 +158,15 @@ def init_repo(path: str, remote: str) -> None:
...
-    cmd_output_b('git', 'remote', 'add', 'origin', remote, cwd=path, env=env)
+
+    def _git(*cmd: str) -> None:
+        cmd_output_b('git', *cmd, cwd=path, env=env)
+
+    os.makedirs(path, exist_ok=True)
+    _git('init', '.')
+    origin = '_pre-commit-origin'
+    _git('remote', 'add', origin, remote)
+    _git('remote', 'rename', origin, 'origin')
```
Comment (webknjaz) [inline pre_commit/git.py:169]: If you're renaming it back, why not use `git config --local` commands as I posted in the issue? Wouldn't it be cleaner?

Comment (asottile) [reply pre_commit/git.py:169]: `git config` to set up a remote feels like a hack, I'd prefer to use `git remote` to set up remotes

**Pull request closed without merging**